packages/agoric-cli/lib/init.js

```diff
@@ -62,6 +62,29 @@ export default async function initMain(_progname, rawArgs, priv, _opts) {
     throw Error('cannot detach from upstream');
   }
 
+  let topLevelName;
+  for (const dir of ['', 'api/', 'contract/', 'ui/']) {
+    const path = `${DIR}/${dir}package.json`;
+    log('rewriting ', path);
+
+    // eslint-disable-next-line no-await-in-loop
+    const contents = await fs.readFile(path, 'utf-8');
+    const pkg = JSON.parse(contents);
+    if (dir === '') {
+      topLevelName = pkg.name;
+    }
+    if (!pkg.name || !pkg.name.startsWith(topLevelName)) {
+      throw Error(
+        `${path}: "name" must start with ${JSON.stringify(topLevelName)}`,
+      );
+    }
+    pkg.name = `${DIR}${pkg.name.substr(topLevelName.length)}`;
+    const json = JSON.stringify(pkg, undefined, 2);
+
+    // eslint-disable-next-line no-await-in-loop
+    await fs.writeFile(path, json);
```

Comment:
Is there some new Node feature I was unaware of that converts Node's callback-based APIs into promise-based APIs without using util.promisify or its ilk?  I.e., is this `await fs.writeFile` a mistake or am I just not up to date on the latest hotness?

Reply:
`fs` is actually `fs.promises`, which does have promise-based APIs.  Perhaps we should rename the power from `fs` to `fsp` or something.